### PR TITLE
[LM-10913] Adding Feedroom support

### PR DESCRIPTION
**Pull request by divyanshgandhilm**
Adding feedroom support in Master SDK.
Improving video handling, and allowing for multiple media types in one post.
Fixing minor bugs, and improving performance.

### CODE REVIEW

File: packages/integration/lib/src/views/feed/topic_select_screen.dart [hunk 1]
```diff
@@ -279,106 +284,191 @@ class _LMFeedTopicSelectScreenState extends State<LMFeedTopicSelectScreen> {
...
-          }
-          return const SizedBox();
-        },
+                    ),
+                  ),
+                ],
+              );
+            }),
+        // builder: (context, state) {
+        //   if (state is LMFeedTopicLoadingState) {
```
Comment (TyagiLikeMinds) [inline packages/integration/lib/src/views/feed/topic_select_screen.dart:373]: can we remove the commented code?

File: packages/integration/lib/src/views/feedroom/feedroom_list_screen_configuration.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
+// part of 'feedroom_screen.dart';
+
+
+// /// {@template feed_screen_config}
+// /// Configuration for the [LMFeedScreen]
+// /// to support changing light and dark style of overall system
```
Comment (TyagiLikeMinds) [inline packages/integration/lib/src/views/feedroom/feedroom_list_screen_configuration.dart:6]: delete this file if not needed

File: packages/integration/lib/src/views/media/media_preview_screen.dart [hunk 3]
```diff
@@ -127,10 +127,10 @@ class _LMFeedMediaPreviewScreenState extends State<LMFeedMediaPreviewScreen> {
               child: CarouselSlider.builder(
                   options: CarouselOptions(
                       initialPage: position ?? 0,
-                      aspectRatio: 9 / 16,
                       enableInfiniteScroll: false,
                       enlargeFactor: 0.0,
                       viewportFraction: 1.0,
+                      aspectRatio: 1,
```
Comment (TyagiLikeMinds) [inline packages/integration/lib/src/views/media/media_preview_screen.dart:133]: Can we get it approved with product once? Doesn't look good with 9:16 images

File: packages/integration/lib/src/widgets/post/comment/comment_reply_widget.dart [hunk 4]
```diff
@@ -503,7 +503,7 @@ class _CommentReplyWidgetState extends State<LMFeedCommentReplyWidget> {
         showDialog(
             context: context,
             builder: (childContext) => LMFeedDeleteConfirmationDialog(
-                title: 'Delete Comment',
+                title: 'Delete Post',
```
Comment (TyagiLikeMinds) [inline packages/integration/lib/src/widgets/post/comment/comment_reply_widget.dart:506]: should be "comment", it is comment reply widget

File: packages/integration/lib/src/views/post/post_detail_screen.dart [hunk 5]
```diff
@@ -446,7 +446,7 @@ class _LMFeedPostDetailScreenState extends State<LMFeedPostDetailScreen> {
           showDialog(
             context: context,
             builder: (childContext) => LMFeedDeleteConfirmationDialog(
-              title: 'Delete Comment',
+              title: 'Delete Post',
```
Comment (TyagiLikeMinds) [inline packages/integration/lib/src/views/post/post_detail_screen.dart:449]: should be "comment" the action is on comment delete

File: packages/integration/pubspec.yaml [hunk 6]
```diff
@@ -33,7 +33,7 @@ dependencies:
   media_kit_video: ^1.2.4 # For video rendering.
   media_kit_libs_video: ^1.0.4
   flutter_keyboard_visibility: ^5.4.1
-  extended_text_field: ^12.0.1
+  extended_text_field: ^14.0.0
```
Comment (TyagiLikeMinds) [inline packages/integration/pubspec.yaml:36]: we should keep the lower bound of package versions

**Review by TyagiLikeMinds — CHANGES_REQUESTED** [review]

File: example/pubspec.yaml [hunk 7]
```diff
@@ -28,7 +28,7 @@ dev_dependencies:
   envied_generator:
   build_runner:
 
-  flutter_lints: ^2.0.0
+  flutter_lints: ^3.0.1
```
[the remark this answers is not in the document]
Comment (divyanshgandhilm) [reply example/pubspec.yaml:31]: This is for example app, doesn't affect the client.

File: packages/integration/lib/src/bloc/feedroom/feedroom_bloc.dart [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+part 'feedroom_event.dart';
+part 'feedroom_state.dart';
+part 'handler/get_feedroom_event_handler.dart';
+part 'handler/get_feedroom_list_event_handler.dart';
+
+class LMFeedRoomBloc extends Bloc<LMFeedRoomEvent, LMFeedRoomState> {
+  // list of selected topics by the user
+  List<LMTopicViewData> selectedTopics = [];
+//   // list of all the topics
+//   Map<String, LMTopicViewData> topics = {};
```
[the remark this answers is not in the document]
Comment (divyanshgandhilm) [reply packages/integration/lib/src/bloc/feedroom/feedroom_bloc.dart:17]: Done

File: packages/integration/lib/src/bloc/feedroom/feedroom_state.dart [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        users,
+        widgets,
+        topics,
+      ];
+}
+
+class LMFeedRoomLoadedState extends LMFeedRoomState {
+  final List<LMPostViewData> posts;
+  final Map<String, LMUserViewData> users;
+  final Map<String, LMWidgetViewData> widgets;
```
[the remark this answers is not in the document]
Comment (divyanshgandhilm) [reply packages/integration/lib/src/bloc/feedroom/feedroom_state.dart:59]: Done

File: packages/integration/lib/src/bloc/post/handler/new_post_event_handler.dart [hunk 10]
```diff
@@ -23,6 +23,7 @@ void newPostEventHandler(
                   : postMedia[0],
         ),
       );
+      try {} catch (e) {}
```
[the remark this answers is not in the document]
Comment (divyanshgandhilm) [reply packages/integration/lib/src/bloc/post/handler/new_post_event_handler.dart:26]: Done

File: packages/integration/lib/src/views/activity/activity_screen.dart [hunk 11]
```diff
@@ -283,11 +283,13 @@ class _LMFeedActivityScreenState extends State<LMFeedActivityScreen> {
         );
       },
       onMediaTap: () async {
-        VideoController? videoController = LMFeedVideoProvider.instance
-            .getVideoController(
-                LMFeedVideoProvider.instance.currentVisiblePostId ?? post.id);
+        // VideoController? videoController = LMFeedVideoProvider.instance
```
[the remark this answers is not in the document]
Comment (divyanshgandhilm) [reply packages/integration/lib/src/views/activity/activity_screen.dart:286]: Done

File: packages/integration/lib/src/views/compose/compose_screen.dart [hunk 12]
```diff
@@ -477,33 +475,37 @@ class _LMFeedComposeScreenState extends State<LMFeedComposeScreen> {
...
-                                  MediaQuery.of(context).size.width - 84,
-                            ) ??
-                            LMFeedPostDocumentStyle(
-                              width: screenSize!.width - 84,
-                              height: 90,
-                            ),
-                        size: PostHelper.getFileSizeString(
-                            bytes: composeBloc.postMedia[index].size ?? 0),
+                      mediaWidget = Padding(
+                        padding: const EdgeInsets.symmetric(horizontal: 18.0),
```
[the remark this answers is not in the document]
Comment (divyanshgandhilm) [reply packages/integration/lib/src/views/compose/compose_screen.dart:479]: It is necessary for current flow

File: packages/integration/lib/src/views/feedroom/feedroom_list_screen_configuration.dart (hunk 2 above)
Comment (divyanshgandhilm) [reply packages/integration/lib/src/views/feedroom/feedroom_list_screen_configuration.dart:6]: Done

**Review by TyagiLikeMinds — APPROVED** [review]

**Review by TyagiLikeMinds — APPROVED** [review]

**Pull request merged**